### ISSUE TITLE
How can I tell how much space and/or how many events are taken up per API Key?

### ISSUE THREAD

**Issue by mason-mcglothlin** [datalust/seq-tickets#648]
We had a massive number of events logged overnight.
![image](https://user-images.githubusercontent.com/8096989/34728775-6a04e2aa-f520-11e7-8ef8-5443a2ff6bef.png)

Is there a way I can tell how much space or how many events are being taken up per API key? Trying to track down the rogue application.

**Comment by nblumhardt** [comment]
Hi Mason, thanks for getting in touch. Yikes, that looks like a bit of a spike!

The _Settings > API Keys_ screen should show counts of events ingested per API key, does that help?

Best regards,
Nick

**Comment by andymac4182** [comment]
@mason-mcglothlin We add a property per API key to see which application caused the log.

**Comment by mason-mcglothlin** [comment]
@nblumhardt Unfortunately, that doesn't help. It only shows the last minute. But as you can see from the pictures above, the traffic died back down before I was able to get to the API Keys screen to check the ingestion rate.

**Comment by nblumhardt** [comment]
@mason-mcglothlin ah, I see - unfortunately I don't think there's a way to tie these back to API keys after-the-fact; @andymac4182 's suggestion will cover it if the issue pops up again.

I don't suppose there are any clues in the events themselves? (Stack traces, maybe?)

**Comment by mason-mcglothlin** [comment]
@andymac4182  Thanks for the idea. Each application in our environment can be uniquely identified by a combination of Application and Environment properties. Once I realized we can run an SQL query and that we can adjust the query timeouts on our preferences page I came up with this:

    select count(*) as NumEvents from stream group by Application, Environment order by NumEvents desc

And that allowed me to identify that our ODS File Ingress application in our Dev 1 environment created 9 million log events. The next closest application/environment had 100K.

Thanks for your assistance!
